nectar/src/database.rs

```diff
@@ -636,3 +643,77 @@ mod tests {
         }
     }
 }
+
+#[cfg(test)]
+mod db_compatibility_tests {
+    use super::*;
+    use crate::database::test_snapshot::v0_1_0;
+    use quickcheck::{Arbitrary, StdThreadGen};
+    use tar::Archive;
+    use tempfile::TempDir;
+
+    const SNAPSHOT_SIZE: usize = 10;
+
+    #[tokio::test]
+    async fn ensure_compatiblity_with_0_1_0() {
+        let tmp_dir = TempDir::new().unwrap();
+        let path = tmp_dir.path();
+
+        let mut ar = Archive::new(&v0_1_0::TAR[..]);
+        ar.unpack(path).unwrap();
+
+        let db = Database::new(path).unwrap();
+
+        let bitcoin_index = db.fetch_inc_bitcoin_transient_key_index().await.unwrap();
+        let stored_swaps = db.all_active_swaps().unwrap();
+        let stored_peers = db.peers().unwrap();
+
+        #[allow(clippy::cast_possible_truncation)]
+        let expected_index = SNAPSHOT_SIZE as u32;
+        assert_eq!(bitcoin_index, expected_index);
+        assert_eq!(stored_swaps.len(), SNAPSHOT_SIZE);
+        assert_eq!(stored_peers.len(), SNAPSHOT_SIZE);
+    }
+
+    #[tokio::test]
+    #[ignore]
```

Comment:
Just an alternative idea:

We could move some code to a `lib.rs` and simply create multiple binaries that use the same code. Then we don't need to (ab)-use ignored tests for this :)